To fix link error

windead opened the pull request:
To fix link error when building //tensorflow/cc:tutorials_example_trainer
The local environment is centos release 6.8 (Final), gcc 4.9.2, bazel 0.5.0.
Error detail: /opt/rh/devtoolset-3/root/usr/bin/ld: bazel-out/local_linux-opt/bin/tensorflow/cc/_objs/tutorials_example_trainer/tensorflow/cc/tutorials/example_trainer.o: undefined reference to symbol 'clock_gettime@@GLIBC_2.2.5'

windead committed:
To fix link error

To fix link error when building //tensorflow/cc:tutorials_example_trainer
The local environment is centos release 6.8 (Final), gcc 4.9.2, bazel 0.5.0.
Error detail: /opt/rh/devtoolset-3/root/usr/bin/ld: bazel-out/local_linux-opt/bin/tensorflow/cc/_objs/tutorials_example_trainer/tensorflow/cc/tutorials/example_trainer.o: undefined reference to symbol 'clock_gettime@@GLIBC_2.2.5'

tensorflow-jenkins commented:
Can one of the admins verify this patch?

martinwicke commented:
Jenkins, test this please.

martinwicke commented:
`-lrt` doesn't work on MacOS. If you can guard this appropriately, it should be fine, there are other platform specific flags in the BUILD files.

See test log for MacOS.

windead committed:
merge the usage of linkopts select

merge the usage of linkopts select from other BUILD file in tf to deal with -lrt in different platform

windead commented:
Commit again. Use the select in linkopts.

martinwicke commented:
Jenkins, test this please.

martinwicke commented:
Thanks!
